src/views/components/survey_components/question_time_setting.vue

```diff
@@ -140,33 +139,39 @@ export default {
             backup_questions: []
         };
     },
-    created: async function() {
+    created: function() {
         this.survey_id = this.$route.query.survey_id;
-        this.name = this.$route.query.name;
+        this.user_id = this.$route.params.user_id;
         // Fetch the list of questions from the API call
-        let fetched = await fetch(
-            "get_info.php?what=surveys&user_id=admin0&survey_id=1"
-        );
-        let fetchedJSON = await fetched.json();
-        // Parse the survey_package returned back into the data
-        if (fetchedJSON.TYPE === "error") {
-            // TODO: redirect to the 404 page
-            return 404;
-        } else if (fetchedJSON.TYPE === "survey_package") {
-            // assign value to the 'num_locked'
+        let url = {
+            what: "surveys",
+            survey_id: this.survey_id,
+            user_id: this.$route.params.user_id
+        };
+
+        fetch("get_info.php?" + generate_query_string(url))
+            .then(res => res.json())
+            .then(data => this.parseData(data))
+            .catch(err => {
+                this.$emit("error", err.toString());
+            });
+    },
+    methods: {
+        parseData: function(fetchedJSON) {
             this.survey_package = fetchedJSON.DATA;
-            if (fetchedJSON.DATA.number_locked_by_department != null) {
+            if (fetchedJSON.DATA[0].number_locked_by_department != null) {
```

Comment:
`if (fetchedJSON.DATA[0] != null && fetchedJSON.DATA[0].number_locked_by_department != null)` on all of these. Otherwise, if `DATA[0]` is undefined, it will throw an error.

Reply:
Better yet, assign `DATA[0]` to a variable and check if it's non-null at the very start.